skabase/SKATestDevice/SKATestDevice/SKATestDevice.py

```diff
@@ -179,6 +213,78 @@ def RunGroupCommand(self, argin):
         return argout
         # PROTECTED REGION END #    //  SKATestDevice.RunGroupCommand
 
+    def devLogMsg(self, devlogmsg, devloglevel):
+
+        if self._element_logging_level >= 1 and devloglevel == 1:
+            self.fatal_stream("%s", devlogmsg)
+        elif self._element_logging_level >= 2 and devloglevel == 2:
+            self.error_stream("%s", devlogmsg)
+        elif self._element_logging_level >= 3 and devloglevel == 3:
+            self.warn_stream("%s", devlogmsg)
+        elif self._element_logging_level >= 4 and devloglevel == 4:
+            self.info_stream("%s", devlogmsg)
+        elif self._element_logging_level >= 5 and devloglevel == 5:
+            self.debug_stream("%s", devlogmsg)
+    #     else:
+    #         pass
+    #
+        if self._central_logging_level >= 1 and devloglevel == 1:
+            self.fatal_stream("%s", devlogmsg)
+        elif self._central_logging_level >= 2 and devloglevel == 2:
+            self.error_stream("%s", devlogmsg)
+        elif self._central_logging_level >= 3 and devloglevel == 3:
+            self.warn_stream("%s", devlogmsg)
+        elif self._central_logging_level >= 4 and devloglevel == 4:
+            self.info_stream("%s", devlogmsg)
+        elif self._central_logging_level >= 5 and devloglevel == 5:
+            self.debug_stream("%s", devlogmsg)
+
+
+        if self._storage_logging_level >= 1 and devloglevel == 1:
+            logger.fatal("%s", devlogmsg)
+        elif self._storage_logging_level >= 2 and devloglevel == 2:
+            logger.error("%s", devlogmsg)
+        elif self._storage_logging_level >= 3 and devloglevel == 3:
+            logger.warn("%s", devlogmsg)
+        elif self._storage_logging_level >= 4 and devloglevel == 4:
+            logger.info("%s", devlogmsg)
+        elif self._storage_logging_level >= 5 and devloglevel == 5:
+            logger.debug("%s", devlogmsg)
+        else:
+            pass
+
+
+
+    @command(
+    )
+    @DebugIt()
+    def On(self):
+        # PROTECTED REGION ID(SKATestDevice.On) ENABLED START #
+
+        self.devLogMsg("TurnOn Sending DEBUG", 5)
```

Comment:
This `devLogMsg` is useful for testing, but do you intend all developers to implement something like this?
The ideal is for the base classes to provide it automatically.

For example, consider a simple device like this:

```python
from skabase import SKABaseDevice

class MyDevice(SKABaseDevice):

    @command(dtype_in='str')
    def DoSomething(self, argin):
        self.logger.info('Received command to do something: %s', argin)

    @command(dtype_in='str')
    def DoSomethingElse(self, argin):
        self.info_stream('Received command to do something else: %s', argin)
```

If we used a DeviceProxy to execute the commands `DoSomething('123')` and `DoSomethingElse('456')`, we would get log lines like:

- `Received command to do something: 123`
- `Received command to do something else:  456`

**Both** lines must be sent to **all** targets:
- syslog
- stdout (console)
- TLS 

I would expect most Python developers to go with the standard `logger.info(...)` type usage, rather than the TANGO `info_stream(...)` usage, but it will be most convenient if we support both.

This requires two parts:
1. adding a log handler to the Python logging facility that forwards the messages to TLS.  Similar to what you've done for syslog.
2. patching the PyTango TLS logging so that calls like `info_stream` go to the standard Python logging.  This is much trickier than number 1.  I'm not sure how exactly we'd do it.  Also need to avoid recursion.

For starters, we should implement number 1.  (Suggest that is  done in a new pull request).  I wouldn't mind dropping the standard `info_stream` type calls for SKA, as these aren't very "Pythonic".

Reply:
We had given a thought on merging Python Logging and TLS. But considering the amount of work involved at this stage, we have parked it for now.
The devlogmsg could be added in SKABaseDevice so there is no need to add in all devices separately.

Reply:
OK, parked for now, but I think implementing it should be high priority.